counsel.el

```diff
@@ -6522,6 +6527,23 @@ list as it may also be a build directory."
         (push (format "%s %s -o %s" cc f o) cmds)))
     cmds))
 
+;; Rust support
+;;
+;; We look for a Cargo.toml and is that exits offer the list of cargo
+;; commands to the user.
+(defun counsel-compile-get-cargo-invocations (&optional dir)
+  "Return a list of cargo targets if we find Cargo.toml."
+  (let ((srcdir (counsel--compile-root))
+        (targets))
+    (when (directory-files (or dir srcdir) nil "Cargo.toml" t)
```

Comment:
The third argument to `directory-files` should actually be `(rx bos "Cargo.toml" eos)`, but better yet why not use `file-exists-p` or similar?